x-pack/plugins/security_solution/public/detection_engine/rule_creation_ui/pages/rule_creation/index.tsx

```diff
@@ -285,16 +289,28 @@ const CreateRulePageComponent: React.FC = () => {
             stepIsValid(scheduleStep) &&
             stepIsValid(actionsStep)
           ) {
-            const createdRule = await createRule(
-              formatRule<RuleCreateProps>(
-                defineStep.data,
-                aboutStep.data,
-                scheduleStep.data,
-                actionsStep.data
-              )
-            );
-
-            displaySuccessToast(i18n.SUCCESSFULLY_CREATED_RULES(createdRule.name), dispatchToaster);
+            const startMlJobsIfNeeded = async () => {
+              if (!isMlRule(defineStep.data.ruleType) || !actionsStep.data.enabled) {
+                return;
+              }
+              setIsStartingJobs(true);
+              await startMlJobs(defineStep.data.machineLearningJobId);
+              setIsStartingJobs(false);
```

Comment:
If `startMlJobs` throws, the `isStartingJobs` state will break. We need to have some error handling in the `startMlJobsIfNeeded` function. Something like that:

```ts
            const startMlJobsIfNeeded = async (): Promise<'ok' | 'failed'> => {
              try {
                if (isMlRule(defineStep.data.ruleType) && actionsStep.data.enabled) {
                  setIsStartingJobs(true);
                  await startMlJobs(defineStep.data.machineLearningJobId);
                }
                return 'ok';
              } catch (e) {
                return 'failed';
              } finally {
                setIsStartingJobs(false);
              }
            };
```

Reply:
It does not throw errors. Everything handled inside. But I can add try/catch wrapping in case you think it will be helpful.

Reply:
done